pkg/stacks/unpack_test.go

```diff
@@ -468,6 +593,31 @@ func TestUnpack(t *testing.T) {
 			root: "ext-dir",
 			want: want{output: expectedSimpleDeploymentStackOutput, err: nil},
 		},
+		{
+			name: "ComplexDeploymentStack",
```

Comment:
do we have test cases/coverage/confidence in the following cases?

* a ui-schema.yaml file at the group level that gets applied to a leaf CRD file because there is no ui-schema.yaml file next to the leaf CRD at it's leaf level?
* a mytype.v1alpha1.ui-schema.yaml file matches to a CRD file of the same type?
* multiple CRDs in the same group each get their own correct corresponding ui-schema.yaml file applied to them? This is probably exactly how our cloud provider stacks will be implemented, so let's make sure this one works.

Some of these may be covered already in this complex test case, but i'm not 💯

Reply:
> a ui-schema.yaml file at the group level that gets applied to a leaf CRD file because there is no ui-schema.yaml file next to the leaf CRD at it's leaf level?

Currently. a group and leaf level CRD would **both** be applied.  This fits my understanding and the `ComplexDeploymentStack` test confirms this behavior.  One of the reasons I believe it should work this way is that we wanted to support multiple vendored ui-schema annotations, so I believe they should all be stacked together.

The order that they are returned is currently alphabetically last, comes first (I could see reason to reverse that).  I'll differentiate the contents of the ui-schema.yaml files in the test so that we can confirm the desired order (whichever way that is).

> a mytype.v1alpha1.ui-schema.yaml file matches to a CRD file of the same type?

I tested a mismatch in `ComplexDeploymentStack`.  I'll add a matching file.

> multiple CRDs in the same group each get their own correct corresponding ui-schema.yaml file applied to them? This is probably exactly how our cloud provider stacks will be implemented, so let's make sure this one works.

"In the same group" -- this should be the case, I'll add a second CRD in this test to confirm.  I'll also add a second CRD in the same leaf directory of one of the other CRDs so we can verify that behavior.

Reply:
[Additional test cases added](https://github.com/crossplaneio/crossplane/compare/017c3d47090daca29f82dfd30b95578a3b3e297d..cd0bf2301c61814fc66c47ea2a3bc4da8c7a2078)